### ISSUE TITLE
Problem With LIKE Query on HBase

### ISSUE THREAD

**Issue by dorianganessa** [Impetus/kundera#889]
Hi, I'm trying to perform a LIKE query with kundera-hbase-v2. I have no errors but it always returns an empty list. I've tried with Criteria and JPQL, nothing seems to work. Other queries with the same methods but without the LIKE clause are working just fine. 

```
            CriteriaBuilder cb = em.getCriteriaBuilder();
            CriteriaQuery<Template> cq = cb.createQuery(Template.class);
            Root<Template> from = cq.from(Template.class);
            cq.select(from.alias("wp"));

            Predicate domainP=null;

            if(domain!=null)
                domainP=cb.like(from.get("rowkey"),"'"+domain+"%'");

            TypedQuery<Template> query = em.createQuery(cq);
            query.getResultList();
```

I tried with and without ' and with and without %, nothing.

As you can see with the Java API this would have been done with a simple scan with startrow and stoprow. Is there any other way to perform a query like this?

**Comment by devender-yadav** [comment]
@dorianganessa 

You can perform `LIKE` query in JPA way. 

Refer [this](https://github.com/impetus-opensource/Kundera/blob/trunk/src/kundera-hbase/kundera-hbase-v2/src/test/java/com/impetus/client/query/LikeQueryTest.java) testcase for reference. Let me know if you face any issue.

-Dev

**Comment by dorianganessa** [comment]
Nothing, I tried with and without named parameter, but no results. Is it possibile that this isn't working because the attribute I'm querying is also the id (rowkey) of the table?

Code I used:

```
String query="Select t from Template t where t.rowkey like :doma";

Query q=em.createQuery(query);
q.setParameter("doma", domain);
return q.getResultList();
```

**Comment by karthikprasad13** [comment]
@dorianganessa 

Like query does not work on rowkey of the table for now. We will fix it asap and add it to the trunk.

-Karthik

**Comment by dorianganessa** [comment]
Ok thank you, it was driving me crazy!
Is there any other way I can perform a query like this with Kundera? Even using hbase shell commands, I don't know

**Comment by devender-yadav** [comment]
Hi @dorianganessa 

Fix has been added in the trunk for this bug and it will be available in the 3.7 release _(expected in a week)_


Thanks for your input.

-Dev
